### PR TITLE
Ra/consumes

### LINKED ISSUE #6295: Add documentation for MVC's [Consumes] attribute
See https://github.com/aspnet/Mvc/issues/7755

We're missing coverage of `[Consumes]` which is an attribute developers can use to limit an action to handling a set of specified content types, or choose a different action based on the request content type.

### PR DESCRIPTION
**Pull request by Rick-Anderson**
@serpent5  I pulled your commits from #16716 here. Please review.

Fixes #6295

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/web-api/index?view=aspnetcore-3.1&branch=pr-en-us-16871#define-supported-request-content-types-with-the-consumes-attribute)

### CODE REVIEW

File: aspnetcore/web-api/index.md [hunk 1]
```diff
@@ -392,6 +392,28 @@ The automatic creation of a `ProblemDetails` instance is disabled when the <xref
 
 ::: moniker-end
 
+## Define supported request content types with the [Consumes] attribute
+
+By default, an action supports all available request content types. For example, if an app is configured to support both JSON and XML [input formatters](xref:mvc/models/model-binding#input-formatters), an action supports multiple content types, including `application/json` and `application/xml`.
```
Comment (Rick-Anderson) [inline aspnetcore/web-api/index.md]: Is by default correct? I didn't understand @serpent5  version which is  "Without further configuration,"

Comment (serpent5) [reply aspnetcore/web-api/index.md]: Yeah, sorry, my wording was clunky. By default seems better to me.

Comment (Rick-Anderson) [reply aspnetcore/web-api/index.md]: @serpent5  really nice work on your part. This seems to be very useful.

File: aspnetcore/web-api/index.md [hunk 2]
```diff
@@ -392,6 +392,28 @@ The automatic creation of a `ProblemDetails` instance is disabled when the <xref
...
+
+The `[Consumes]` attribute also allows an action to influence its selection based on an incoming request's content type by applying a type constraint. Consider the following example:
+
+[!code-csharp[](index/samples/3.x/Controllers/ConsumesController.cs?name=snippet_Class)]
+
+In the preceding code, `ConsumesController` is configured to handle requests sent to the `https://localhost:5001/api/Consumes` URL. Both of the controller's actions, `PostJson` and `PostForm`, handle POST requests with the same URL. Without the `[Consumes]` attribute applying a type constraint, an ambiguous match exception is thrown.
+
+The `[Consumes]` attribute is applied to both actions. The `PostJson` action handles requests sent with a `Content-Type` header of `application/json` The `PostForm` action handles requests sent with a `Content-Type` header of `application/x-www-form-urlencoded`.
```
Comment (serpent5) [inline aspnetcore/web-api/index.md]:
```suggestion
The `[Consumes]` attribute is applied to both actions. The `PostJson` action handles requests sent with a `Content-Type` header of `application/json`. The `PostForm` action handles requests sent with a `Content-Type` header of `application/x-www-form-urlencoded`. 
```

Missing `.` to end the second sentence.

File: aspnetcore/web-api/index.md [hunk 3]
```diff
@@ -392,6 +392,28 @@ The automatic creation of a `ProblemDetails` instance is disabled when the <xref
 
 ::: moniker-end
 
+## Define supported request content types with the [Consumes] attribute
+
+By default, an action supports all available request content types. For example, if an app is configured to support both JSON and XML [input formatters](xref:mvc/models/model-binding#input-formatters), an action supports multiple content types, including `application/json` and `application/xml`.
+
+The [`[Consumes]`](<xref:Microsoft.AspNetCore.Mvc.ConsumesAttribute>) attribute allows an action to limit the supported request content types. Apply the `[Consumes]` attribute to an action or controller, specifying one or more content types:
```
Comment (scottaddie) [inline aspnetcore/web-api/index.md]:
```suggestion
The [[Consumes]](<xref:Microsoft.AspNetCore.Mvc.ConsumesAttribute>) attribute allows an action to limit the supported request content types. Apply the `[Consumes]` attribute to an action or controller, specifying one or more content types:
```

File: aspnetcore/web-api/index.md [hunk 4]
```diff
@@ -392,6 +392,28 @@ The automatic creation of a `ProblemDetails` instance is disabled when the <xref
...
+
+```csharp
+[HttpPost]
+[Consumes("application/xml")]
+public IActionResult CreateProduct(Product product)
+```
+
+In the preceding code, the `CreateProduct` action specifies the content type  `application/xml`. Requests routed to this action must specify a `Content-Type` header of `application/xml`. Requests that do not specify a `Content-Type` header of `application/xml` result in a [415 Unsupported Media Type](https://developer.mozilla.org/docs/Web/HTTP/Status/415) response.
```
Comment (scottaddie) [inline aspnetcore/web-api/index.md]:
```suggestion
In the preceding code, the `CreateProduct` action specifies the content type `application/xml`. Requests routed to this action must specify a `Content-Type` header of `application/xml`. Requests that don't specify a `Content-Type` header of `application/xml` result in a [415 Unsupported Media Type](https://developer.mozilla.org/docs/Web/HTTP/Status/415) response.
```

**Pull request merged**